print_minute.c

```diff
@@ -46,12 +47,14 @@ int main(int argc, char * argv[]) {
     time_t foo = time(NULL);
     struct tm *tm_struct = localtime(&foo);
     int newMinute = tm_struct->tm_min;
+
     if (minute != newMinute) {
         read_file_line_by_line(newMinute);
         system(command);
-        execv(argv[0], argv);
     } else {
-        printf("%d\n", minute);
+        printf("%d\nSleeping for 10 seconds.\n", minute);
+        sleep(tenSeconds);
     }
-    return EXIT_SUCCESS;
+
+    return execv(argv[0], argv);
```

Comment:
I don't have a great amount of expertise with c, does this mean that new processes will eventually shut down/exit?

Reply:
AFAIK know yes, C uses eager execution, so it would be same as 

```bash
"$0"
exit $?
```

Reply:
execv never returns normally. It replaces the current process with a new one, the same as bash's exec. So there will only ever be one copy of the application running.

If exec fails (very unlikely) we return the return code of exec.